app/validators/rules/rule_validator.py

```diff
@@ -262,3 +271,97 @@ def _validate_options(self, rules, operator_name):
                         value=value,
                         answer_options=list(option_values),
                     )
+
+    def _validate_operator_arguments(self, rule):
+        operator_name = next(iter(rule))
+        argument_types = self._get_argument_types_for_operator(rule[operator_name])
+
+        if operator_name in COMPARISON_OPERATORS + ARRAY_OPERATORS + NUMERIC_OPERATORS:
+            self._validate_comparison_operator_argument_types(
+                rule, operator_name, argument_types
+            )
+
+        if (
+            operator_name in COMPARISON_OPERATORS + [Operator.ALL_IN, Operator.ANY_IN]
+            and TYPE_NULL not in argument_types
+        ):
+            self._validate_argument_types_match(rule, argument_types)
+
+        if operator_name == Operator.DATE:
+            return TYPE_DATE
+
+        if operator_name == Operator.COUNT:
+            return TYPE_NUMBER
+
+        return TYPE_BOOLEAN
```

Comment:
Need one for `NUMERIC` operators instead of COUNT.

Reply:
Updated 👍